lib/ruby_lsp/requests/code_lens.rb

```diff
@@ -44,17 +45,52 @@ def run
       def visit_class(node)
         class_name = node.constant.constant.value
         if class_name.end_with?("Test")
-          add_code_lens(node, [@path, class_name, BASE_COMMAND + " -Itest " + @path])
+          add_code_lens(node, path: @path, name: class_name, command: BASE_COMMAND + " -Itest " + @path)
+        end
+        visit(node.bodystmt)
+      end
+
+      sig { override.params(node: SyntaxTree::DefNode).void }
+      def visit_def(node)
+        if @modifier == "public"
+          method_name = node.name.value
+          if method_name.start_with?("test")
+            add_code_lens(
+              node,
+              path: @path + ":" + node.location.start_line.to_s,
+              name: method_name,
+              command: BASE_COMMAND + " -Itest " + @path + " --name " + method_name,
```

Comment:
We should probably treat `method_name` as a regexp something like `%r{^{method_name}}` otherwise this may be run the wrong test if you have something like `test_foo` and `test_foo_another` where one is a substring of the other.

Reply:
correction: `%r{^#{method_name}$}`

Reply:
Yeah, that one is just so that we don't have to type the entire name when running manually.